buildSrc/src/main/java/org/elasticsearch/gradle/precommit/NamingConventionsTask.java

```diff
@@ -33,24 +32,17 @@ public NamingConventionsTask() {
 
         SourceSetContainer sourceSets = getJavaSourceSets();
         final FileCollection classpath;
-        try {
-            URL location = NamingConventionsCheck.class.getProtectionDomain().getCodeSource().getLocation();
-            if (location.getProtocol().equals("file") == false) {
-                throw new GradleException("Unexpected location for NamingConventionCheck class: "+ location);
-            }
-            classpath = project.files(
-                    // This works because the class only depends on one class from junit that will be available from the
-                    // tests compile classpath. It's the most straight forward way of telling Java where to find the main
-                    // class.
-                    location.toURI().getPath(),
-                    // the tests to be loaded
-                    checkForTestsInMain ? sourceSets.getByName("main").getRuntimeClasspath() : project.files(),
-                    sourceSets.getByName("test").getCompileClasspath(),
-                    sourceSets.getByName("test").getOutput()
-            );
-        } catch (URISyntaxException e) {
-            throw new AssertionError(e);
-        }
+        classpath = project.files(
+                // This works because the class only depends on one class from junit that will be available from the
+                // tests compile classpath. It's the most straight forward way of telling Java where to find the main
+                // class.
+                ClassPathUtils.getJar(NamingConventionsCheck.class).toFile(),
```

Comment:
I would prefer if we kept this url hack as it was and didn't make it easier to use